frontend/packages/ceph-storage-plugin/src/__tests__/independent-mode/breakdown-card.spec.tsx

```diff
@@ -0,0 +1,68 @@
+
+import { DashboardItemProps } from '@console/internal/components/dashboard/with-dashboard-resources';
+import { Dropdown } from '@console/internal/components/utils';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+
+import { BreakdownCard } from '../../components/independent-dashboard-page/breakdown-card/card';
+import { BreakdownCardBody } from '../../components/dashboard-page/storage-dashboard/breakdown-card/breakdown-body';
+import {
+  DashboardItems,
+  dropDownItems,
+  headerLink,
+} from '../../__mocks__/independent-mode-dashboards';
+import { HeaderPrometheusViewLink } from '../../components/dashboard-page/storage-dashboard/breakdown-card/breakdown-header';
+
+describe('BreakdownCard', () => {
+  let wrapper: ShallowWrapper<DashboardItemProps>;
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <BreakdownCard
+        watchPrometheus={DashboardItems.watchPrometheus as any}
+        stopWatchPrometheusQuery={DashboardItems.stopWatchPrometheusQuery as any}
+        prometheusResults={DashboardItems.prometheusResults as any}
+        watchURL={DashboardItems.watchURL as any}
+        stopWatchURL={DashboardItems.stopWatchURL as any}
+        watchAlerts={DashboardItems.watchAlerts as any}
+        stopWatchAlerts={DashboardItems.stopWatchAlerts as any}
+        urlResults={DashboardItems.urlResults as any}
+        notificationAlerts={DashboardItems.notificationAlerts as any}
+        watchK8sResource={DashboardItems.watchK8sResource as any}
+        stopWatchK8sResource={DashboardItems.stopWatchK8sResource as any}
+      />,
+    ).dive();
+  });
+
+  it('Should render Card Header', () => {
+    expect(wrapper.find(DashboardCardHeader).exists()).toBe(true);
+  });
+
+  it('Should render Card Title', () => {
+    expect(wrapper.find(DashboardCardTitle).exists()).toBe(true);
+  });
+
+  it('Should render Prometheus Header Link', () => {
+    expect(wrapper.find(HeaderPrometheusViewLink).exists()).toBe(true);
+    expect(wrapper.find(HeaderPrometheusViewLink).props().link).toEqual(headerLink);
+  });
+
+  it('Should render Dropdown', () => {
+    expect(wrapper.find(Dropdown).exists()).toBe(true);
+    expect(wrapper.find(Dropdown).props().items).toEqual(dropDownItems);
+    expect(wrapper.find(Dropdown).props().title).toEqual('Projects');
+  });
+
+  it('Should render Card body', () => {
+    expect(wrapper.find(DashboardCardBody).exists()).toBe(true);
+  });
+
+  it('Should render Breakdown Card body', () => {
+    expect(wrapper.find(BreakdownCardBody).exists()).toBe(true);
+    expect(wrapper.find(BreakdownCardBody).props().isLoading).toBe(true);
+    expect(wrapper.find(BreakdownCardBody).props().hasLoadError).toBe(false);
+  });
+});
```

Comment:
These tests are part of Breakdown Card right? Why are we testing them again? We should only test whether Breakdown card is getting rendered or not. If these tests are not there in Breakdown lets add them there rather than here.

Reply:
@bipuladh these tests are for breakdown card of independent mode. All these component are part of the card directly, looking at the [component](https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/independent-dashboard-page/breakdown-card/card.tsx). 

I believe test related to how breakdown card uses props should be part of the breakdown-card test. But, as part of this card, we should add test related to expected props been sent to breakdown component, they are set and calculated here.

Reply:
> These tests are part of Breakdown Card right?

It seems that `BreakdownCard` component tested here is OCS / independent mode specific. The code is currently in `src/components/independent-dashboard-page/breakdown-card/card.tsx`.

Similarly, there is e.g. (OCS specific) `UtilizationCard` in `src/components/independent-dashboard-page/utilization-card/card.tsx`.

Both of these are built on top of `DashboardCard` which is a core Console component.

What we're missing here (I think) is a test for the core `DashboardCard` component:

```
git grep 'DashboardCard' *.spec.*
# no results
```

For now, I'm OK with these tests being here. In future, we should refactor & move relevant bits into ^^ spec.

cc @rawagner

Reply:
Once we have a core `DashboardCard` spec, tests for OCS specific components built on top of it should only test the OCS specifics (in terms of their render output or behavior).